src/ILVerify/tests/ILMethodTester.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Internal.IL;
+using Internal.TypeSystem.Ecma;
+using Xunit;
+
+namespace ILVerify.Tests
+{
+    public class ILMethodTester
+    {
+        [Theory]
+        [MemberData(nameof(TestDataLoader.GetMethodsWithValidIL), MemberType = typeof(TestDataLoader))]
+        void TestMethodsWithValidIL(ValidILTestCase validILTestCase)
+        {
+            ILImporter importer = ConstructILImporter(validILTestCase);
+
+            var verifierErrors = new List<VerifierError>();
+            importer.ReportVerificationError = new Action<VerificationErrorArgs>((err) =>
+            {
+                verifierErrors.Add(err.Code);
+            });
+
+            importer.Verify();
+            Assert.Equal(0, verifierErrors.Count);
+        }
+
+        [Theory]
+        [MemberData(nameof(TestDataLoader.GetMethodsWithInvalidIL), MemberType = typeof(TestDataLoader))]
+        void TestMethodsWithInvalidIL(InvalidILTestCase invalidILTestCase)
+        {
+            ILImporter importer = ConstructILImporter(invalidILTestCase);
+
+            var verifierErrors = new List<VerifierError>();
+            importer.ReportVerificationError = new Action<VerificationErrorArgs>((err) =>
+            {
+                verifierErrors.Add(err.Code);
+            });
+            
+            try
+            {
+                importer.Verify();
+            }
+            catch
+            {
+                //in some cases ILVerify throws exceptions when things look too wrong to continue
```

Comment:
Could you please open the issue on defining the public surface and add a link to it here?